### PR TITLE
Filter refactoring opportunities for the files open in the Editor

### PR DESCRIPTION
**Pull request by foiki**
…ses for refactoring

### CODE REVIEW

File: src/main/java/ui/SettingsPanel.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    private static Set<ClassObject> getAllOpenedClassObjects() {
+        List<ClassObject> classObjectsToBeExamined = new ArrayList<>();
+        VirtualFile[] currentOpenedFiles = FileEditorManager.getInstance(project).getOpenFiles();
+        for (VirtualFile virtualFile : currentOpenedFiles) {
+            PsiFile psiFile = PsiManager.getInstance(project).findFile(virtualFile);
+            if (psiFile != null) {
+                for (PsiClass psiClass : ((PsiJavaFile) psiFile).getClasses()) {
+                    classObjectsToBeExamined.add(new ClassObject(psiClass));
```
Comment (onewhl) [inline src/main/java/ui/SettingsPanel.java]: Are you sure that we need to create a new `ClassObject` instance here and in `getCurrentOpenedClassObject()`? We already created `ClassObject` instance for each `PsiClass` inside `ASTReader` constructor.

File: src/main/java/core/ast/ASTReader.java [hunk 2]
```diff
@@ -27,6 +27,7 @@ public ASTReader(ProjectInfo project, ProgressIndicator indicator) {
         for (PsiClass c : classes) {
             systemObject.addClass(processTypeDeclaration(c));
             processedClasses += 1;
+            indicator.setIndeterminate(false);
```
Comment (onewhl) [inline src/main/java/core/ast/ASTReader.java]: Why did you add this line? I think we should set `true` here.

File: src/main/java/ui/SettingsPanel.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                }
+            }
+        }
+        System.out.println(classObjectsToBeExamined.size());
+        return new LinkedHashSet<>(classObjectsToBeExamined);
+    }
+
+    private static Set<ClassObject> getCurrentOpenedClassObject() {
```
Comment (onewhl) [inline src/main/java/ui/SettingsPanel.java]: Rename to `getCurrentOpenedPsiClasses()`

File: src/main/java/ui/SettingsPanel.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        Set<ClassObject> classObjectsToBeExamined = new LinkedHashSet<>();
+        FileEditor currentEditor = FileEditorManager.getInstance(project).getSelectedEditor();
+        if (currentEditor != null && currentEditor.getFile() != null) {
+            PsiClass psiClass = PsiUtils.findClass(currentEditor.getFile().getName(), project);
+            if (psiClass != null) {
+                classObjectsToBeExamined.add(new ClassObject(psiClass));
```
Comment (onewhl) [inline src/main/java/ui/SettingsPanel.java]: Please, clarify the global difference between `getAllOpenedClassObjects()` and `getCurrentOpenedClassObject().`

File: src/main/java/org/jetbrains/research/intellijdeodorant/ide/ui/SettingsPanel.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        final JBPanel panel = new JBPanel();
+        final JBPanel componentPanel = new JBPanel<JBPanel<JBPanel>>();
+        panel.setLayout(new FlowLayout(FlowLayout.LEFT));
+        componentPanel.add(new Label(IntelliJDeodorantBundle.message(REFACTORING_PLACE_CHOOSE_BOX)));
+        componentPanel.add(refactoringPlaceChooseBox);
+        panel.add(componentPanel, BorderLayout.WEST);
+        return panel;
+    }
+
+    public static Set<ClassObject> getClassesToFindRefactorings(SystemObject systemObject) {
```
Comment (123a6bcw) [inline src/main/java/org/jetbrains/research/intellijdeodorant/ide/ui/SettingsPanel.java]: I suggest creating two methods -- one that returns `Set<PsiClass>`, that way we could use it for compilation check (and compile the whole project), and another - that returns `Set<ClassObject>` (apply `ASTReader.processTypeDeclaration` to the result of the first method).

**Comment by onewhl** [comment]
Implemented in 2020.1-1.1
